sdks/python/apache_beam/coders/coder_impl.py

```diff
@@ -510,6 +510,9 @@ def encode_special_deterministic(self, value, stream):
       stream.write_byte(NESTED_STATE_TYPE)
       self.encode_type(type(value), stream)
       state_value = value.__getstate__()
+      if value is not None and state_value is None:
```

Comment:
Why are we checking that value is not `None` here? I don't think we want to use this protocol for `None`.

Reply:
This just indicates `__getstate__` might not be implemented. And we know this is true for this if checks.